dashboard.py

```diff
@@ -0,0 +1,177 @@
+from flask import jsonify, render_template, request, make_response, Blueprint, url_for, session, redirect
+import json
+from utils.make_session import make_session
+import rethinkdb as r
+import hashlib
+from random import randint
+
+config = json.load(open('config.json'))
+
+dash = Blueprint('dashboard', __name__, template_folder='views', static_folder='views/assets')
+
+API_BASE_URL = 'https://discordapp.com/api'
+AUTHORIZATION_BASE_URL = API_BASE_URL + '/oauth2/authorize'
+TOKEN_URL = API_BASE_URL + '/oauth2/token'
+RDB_ADDRESS = config['rdb_address']
+RDB_PORT = config['rdb_port']
+RDB_DB = config['rdb_db']
+
+rdb = r.connect(RDB_ADDRESS, RDB_PORT)
+
+
+@dash.route('/login')
+def login():
+    discord = make_session(scope='identify email', redirect_uri=request.host_url + 'callback')
+    authorization_url, state = discord.authorization_url(AUTHORIZATION_BASE_URL)
+    return redirect(authorization_url)
+
+
+@dash.route('/logout')
+def logout():
+    session.clear()
+    return redirect(request.host_url)
+
+
+@dash.route('/callback')
+def callback():
+    print(url_for('.callback'))
+    if request.values.get('error'):
+        return request.values['error']
+    discord = make_session(state=session.get('oauth2_state'), redirect_uri=request.host_url + 'callback')
+    token = discord.fetch_token(
+        TOKEN_URL,
+        client_secret=config['client_secret'],
+        authorization_response=request.url)
+    session['oauth2_token'] = token
+    return redirect(url_for('.dashboard'))
+
+
+@dash.route('/dashboard')
+def dashboard():
+    discord = make_session(token=session.get('oauth2_token'))
+    user = discord.get(API_BASE_URL + '/users/@me').json()
+    try:
+        user['id']
```

Comment:
As the result is not used here, I think it'd probably better to do `if id not in user`. It'd also shorten code by ~2 lines